tutorials/core-concepts/pieces/pieces-example.md

```diff
@@ -0,0 +1,242 @@
+```
+
+## Create the Piece Widget
+
+Next, define the template for the widget to display the new pieces.
+
+1. Open `lib/modules/menu-item-widgets/views/widget.html`
+2. Add the following code to define the display template for the piece:
+  ```markup
+    {% for piece in data.widget._pieces %}
+
+      <h4>{% if piece._url %}<a href="{{ piece._url }}">{% endif %}
+
+      {{ piece.title }}
+
+      {% if piece._url %}</a>{% endif %}</h4>
+
+      ${{ piece.price }}
+
+      <p>{{ piece.description }}</p>
+
+      {{ apos.singleton( piece, 'image', 'apostrophe-images',
+
+        { edit: false }
+
+      )}}
+
+    {% endfor %}
+  ```
+
+## Displaying Pieces with a Widget
+
+Next, create some pieces to display and add the new widget to a page.
+
+### Create Pieces
+
+1. Log in to your Apostrophe instance.
+
+2. Open the main menu and select **Food**
+
+3. Create some food to be displayed.
+
+![](/.gitbook/assets/pieces-manage-food.png) 
+
+### Add the Widget to a Page
+
+1. Open `lib/modules/apostrophe-pages/views/pages/home.html`
+
+2. Add the widget to the template with an area
+
+  ```markup
+    {{ apos.area(data.page, 'body', {
+      widgets: {
+        'foods': {}
+        }
+      }
+    )}}
+  ```
+
+3. Navigate back to the home page, and you can now add a Menu Item Widget to the page.
+
+4. Click *Add Food Widget* and you can define which pieces to display and how to display them in the widget.
+
+![](/.gitbook/assets/pieces-add-food-widget.png) 
+
+```
+Note: in the widget template, you defined a **for loop** which looks through the list of all pieces to display. When you display the pieces in the widget, the selection you make in the dialog box will define the complete list of pieces that the template sees.
+```
+
+![](/.gitbook/assets/pieces-widget-page.png) 
+
+Congratulations! You created a new content type and a widget to display it. However, there are some limitations to using a widget to display this sort of content. One limitation is that everything you do and configure is contained within the instance of the widget placed on a specific page. Another is that the pieces themselves don’t have a “home” that you can universally link to them from. Using pieces pages will solve this.
+
+## Create the Pieces Pages
+
+Since you created the new piece type with the `--pages` options, you already have the pieces pages module created. Now, you must register the pieces pages module in `app.js`, and create the templates for the index and show pages. 
+
+![](/.gitbook/assets/pieces-pages-directory.png) 
+
+1. Add the following code to `app.js` in the main modules section to register the pieces pages module.
+  
+  ```markup
+    modules { 
+      ... 
+      'foods-pages': { 
+          extend: 'apostrophe-pieces-pages'
+        }, 
+      ...
+    }
+  ```
+
+2. Then register the pieces pages as a new page type by adding this is the `pages` subsection of `modules`:
```

Comment:
Grammar. Also, `apostrophe-pages` is the module name, not `pages`. Please review this section for correctness.